src/updater/updater.cpp

```diff
@@ -583,8 +583,15 @@ Updater::LoadAndUpdateEdgeExpandedGraph(std::vector<extractor::EdgeBasedEdge> &e
             [&] { extractor::files::readNodeData(config.GetPath(".osrm.ebg_nodes"), node_data); },
 
             [&] {
-                extractor::files::readTurnWeightPenalty(
-                    config.GetPath(".osrm.turn_weight_penalties"), turn_weight_penalties);
+                // we are going to overwrite this `.turn_weight_penalties` afterwards,
+                // so here we backup the original turn penalties if we didn't do that yet in order to
+                // guarantee that subsequent runs of this code will work on top of original weights
+                auto path = config.GetPath(".osrm.turn_weight_penalties").string() + ".original";
```

Comment:
Also this problem made me think about if we can handle all these time based restrictions automatically without forcing users to make error-prone operations to handle this. It feels it should be relatively not so many time based restrictions in the graph and we probably can just handle it dynamically. It seems to be a bit against philosophy of OSRM when it tries to precalculate all weights to be as fast as possible, but at the same time it should add a lot of flexibility:
- time based restrictions will "just work"  without additional calls with `--parse-conditionals-from-now`
- we potentially will be able to implement "depart at" feature

Reply:
Can probably be discussed separately. There will be practical as well as philosophical issues.